rclpy/src/rclpy_common/include/rclpy_common/handle.h

```diff
@@ -0,0 +1,116 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLPY_COMMON__HANDLE_H_
+#define RCLPY_COMMON__HANDLE_H_
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include <Python.h>
+#include <stddef.h>
+
+#include "rcutils/types/rcutils_ret.h"
+
+/// Wrapper that manages the lifetime of an object,
+/// allowing to establish dependencies between them.
+typedef struct rclpy_handle_t rclpy_handle_t;
+
+/// Signature of rclpy_handle_t destructors.
+typedef void (* rclpy_handle_destructor_t)(void *);
+
+/// Create a PyCapsule wrapping a rclpy_handle_t object.
+/**
+ * Handles are used to manage lifetime of rcl objects.
```

Comment:
@ivanpauno I'd move this rationale to the `typedef` above.

Reply:
I deleted this.
The above typedef already mentioned this rationale.